docs/security/safe-storage-app-secrets.rst

```diff
@@ -0,0 +1,132 @@
+
+Apps frequently require passwords, such as a client ID for OAuth. These passwords and other sensitive data should never be added to *config* files inside your projects source tree, as configuration files can be accidentally checked into source code. The secrets manager tool provides a  mechanism to store sensitive data for development work outside your project tree. The secrets manager tool is a DNX console application that is used to store secrets used by ASP.NET applications in the development environment.
+
+Installing the secrets manager tool
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Create a new ASP.NET web app. We will use this to test secrets stored with the secrets manager tool tool.
+- Open a command prompt and navigate to the project folder (the folder with *config.json*).
+- Set the runtime version using the .Net version manager(DNVM). DNVM is a tool that lets you list, install and switch `DNX <http://docs.asp.net/en/latest/dnx/overview.html?highlight=dnvm>`__ versions on your machine. Run the following command:
+
+.. TODO each version, update version # (1.0.0-beta4)
+	note: there are multiple versions in this file
+
+.. code-block:: none
+
+    dnvm use 1.0.0-beta4
+    
+- Install the secrets manager tool using DNU (Microsoft .NET Development Utility). DNU is used to build, package and publish DNX projects.
+ 
+.. code-block:: none
+ 
+    dnu commands install SecretManager
+    
+With Visual Studio 2015 RC, you will get the following error:
+
+.. code-block:: none
+
+    Unable to locate SecretManager >= 1.0.0-beta4-10173
+
+To get around this error, we will remove the "-10173" version number. Open  *C:\\Users\\<username>\\.dnx\\bin\\packages\\SecretManager\\1.0.0-beta4\\app\\project.json* and find the following line:
+
+ **"SecretManager": "1.0.0-beta4-10173"**
+ 
+ .. This step should not be necessary in the next version
+
+Remove  "-10173". The completed markup is shown below.
+
+.. code-block:: json
+
+		{
+		  "version": "1.0.0-*",
+		  "description": "ASP.NET 5 tool to manage user secrets.",
+		  "dependencies": {
+			"SecretManager": "1.0.0-beta4"
+		  },
+  
+Save the file and run the dnu command again, it should successfully install and report "The following commands were installed: user-secret". The secrets manager tool is now installed globally. Close the command window.
+
+- Open a new command window and enter the following:
+
+.. code-block:: none
+
+	dnvm use default
+
+**dnvm** is the .NET Version Manager, a set of command line utilities that are used to update and configure .NET Runtime. The command ``dnvm use default`` instructs the .NET Version Manager to add the ASP.NET 5 runtime to the ``PATH`` environment variable for the current shell. For Visual Studio 2015 RC, the following is displayed: 

```

Comment:
`update and configure .NET Runtime` --> `update and configure the .NET Execution Environment`